starlette_admin/fields.py

```diff
@@ -1267,3 +1272,57 @@ def additional_css_links(
 
     def additional_js_links(self, request: Request, action: RequestAction) -> List[str]:
         return self.field.additional_js_links(request, action)
+
+
+@dataclass
+class IntervalField(BaseField):
+    render_function_key: str = "interval"
+    form_template: str = "forms/interval.html"
+    display_template: str = "displays/interval.html"
+
+    async def parse_form_data(
+        self, request: Request, form_data: FormData, action: RequestAction
+    ) -> Any:
+        timedelta_params = {
+            "weeks": (
+                0
+                if form_data.get(f"{self.name}_weeks") == ""
```

Comment:
You should use `self.id` here so that the field can be used with ListField and CollectionField.
```suggestion
                if form_data.get(f"{self.id}_weeks") == ""
```

Reply:
same for other attributes